update_spreadsheet.py

```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+import os
+import pytz
+
+from datetime import datetime
+from oauth2client import client
+from oauth2client import tools
+from oauth2client.file import Storage
+
+try:
+    import argparse
+    flags = argparse.ArgumentParser(parents=[tools.argparser]).parse_args()
+except ImportError:
+    flags = None
+
+
+class UpdateSpreadsheet(object):
+    tokyo_timezone = pytz.timezone('Asia/Tokyo')
+    utc = datetime.now()
+    jst = tokyo_timezone.fromutc(utc)
+
+    def __init__(self, env):
+        self.env = env
+
+    def get_credentials(self):
+        if not os.path.exists(self.env.CREDENTIAL_DIR):
+            os.makedirs(self.env.CREDENTIAL_DIR)
+
+        credential_path = os.path.join(self.env.CREDENTIAL_DIR, self.env.CREDENTIAL_JSON)
+
+        store = Storage(credential_path)
+        credentials = store.get()
+        if not credentials or credentials.invalid:
+            flow = client.flow_from_clientsecrets(self.env.CLIENT_SECRET_JSON, self.env.SCOPES)
+            flow.user_agent = self.env.APPLICATION_NAME
+            if flags:
+                credentials = tools.run_flow(flow, store, flags)
+            else:
+                credentials = tools.run_flow(flow, store)
+            print('Storing credentials to ' + credential_path)
+        return credentials
+
+    def get_worksheet(self, gc):
+        date = datetime.now().strftime('%Y-%m')
+
+        workbook = gc.open_by_key(self.env.SPREADSHEET_KEY)
+        worksheet = workbook.worksheet(date)
+
+        return worksheet
+
+    def update_in_time(self, worksheet):
```

Comment:
```suggestion
    def update_in_time(self, worksheet, now=datetime.now()):
```

「現在時刻」を関数の中で取得するのは避けたほうがいい。外部から与えるようにするとテスト実行などの際に便利。

Reply:
78dc0ef
set_jst_now()というstaticmethod内で現在時刻をreturnして
インスタンス変数に渡すようにしてみましたがどうでしょうか？